stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/repository/MessageRepository.kt

```diff
@@ -102,13 +96,14 @@ internal class MessageRepository(
         messageCache.remove(message.id)
     }
 
-    internal suspend fun selectSyncNeeded(getUser: suspend (userId: String) -> User): List<Message> {
-        return messageDao.selectSyncNeeded().map { it.toModel(getUser) { select(it, getUser) } }
+    suspend fun selectMessagesSyncNeeded(): List<Message> {
+        return messageDao.selectSyncNeeded().map { it.toModel(getUser, ::selectMessage) }
     }
 
-    companion object {
+    private companion object {
         private const val DEFAULT_MESSAGE_LIMIT = 100
+
+        private fun allMessages(message: Message): List<Message> =
```

Comment:
Is it placed inside `companion object` on purpose?

Reply:
Yes, because it was placed as an extension, but no other clients used it. So obviously it's an extension method only for this class and I put it in companion